core/domain/src/main/java/com/mateuszholik/domain/usecases/GetHead2HeadUseCase.kt

```diff
@@ -0,0 +1,21 @@
+package com.mateuszholik.domain.usecases
+
+import com.mateuszholik.common.providers.DispatchersProvider
+import com.mateuszholik.data.repositories.MatchesRepository
+import com.mateuszholik.domain.usecases.base.ParameterizedFlowUseCase
+import com.mateuszholik.model.Head2Head
+import com.mateuszholik.model.Result
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.flowOn
+
+interface GetHead2HeadUseCase : ParameterizedFlowUseCase<Int, Head2Head>
+
+internal class GetHead2HeadUseCaseImpl(
+    private val matchesRepository: MatchesRepository,
+    private val dispatchersProvider: DispatchersProvider,
+) : GetHead2HeadUseCase {
+
+    override fun invoke(param: Int): Flow<Result<Head2Head>> =
+        matchesRepository.getMatchH2H(param)
+            .flowOn(dispatchersProvider.io)
```

Comment:
So that's the case I was referring to in previous PR. You don't have to put it after every flow. Maybe it's better to put close to the end (so here) or change to Default, because it's no longer connected with IO